src/benchq/resource_estimation/graph/estimators.py

```diff
@@ -65,32 +68,41 @@ def _logical_cell_failure_rate(self, distance: int) -> float:
     def _minimize_code_distance(
         self,
         n_nodes: int,
+        max_graph_degree: int,
         error_budget,
-        error_rate: Callable[[int, int], float],
+        error_rate: Callable[[int, int, int], float],
         min_d: int = 4,
         max_d: int = 100,
     ) -> int:
         target_error_rate = error_budget["total_error"] * error_budget["ec_error_rate"]
 
         for distance in range(min_d, max_d):
-            if error_rate(distance, n_nodes) < target_error_rate:
+            if error_rate(distance, n_nodes, max_graph_degree) < target_error_rate:
                 return distance
 
         raise RuntimeError(f"Not found good error rates under distance code: {max_d}.")
 
-    def _ec_error_rate_synthesized(self, distance: int, n_nodes: int) -> float:
+    def _ec_error_rate_synthesized(
+        self, distance: int, n_nodes: int, max_graph_degree: int
+    ) -> float:
         return self._logical_cell_failure_rate(distance) * self.get_logical_st_volume(
-            n_nodes
+            n_nodes, max_graph_degree
         )
 
-    def _ec_error_rate_unsynthesized(self, distance: int, n_nodes: int) -> float:
+    def _ec_error_rate_unsynthesized(
+        self, distance: int, n_nodes: int, max_graph_degree: int
+    ) -> float:
         _, ec_error_rate = self.balance_logical_error_rate_and_synthesis_accuracy(
-            n_nodes, distance
+            n_nodes, distance, max_graph_degree
         )
         return ec_error_rate
 
-    def get_logical_st_volume(self, n_operations):
-        return 12 * n_operations * self.N_TOCKS_PER_T_GATE_FACTORY * n_operations
+    def get_logical_st_volume(self, n_nodes: int, max_graph_degree: int):
+        num_boxes = np.ceil((max_graph_degree + 1) / 21)
+        space = 21 * 12 * num_boxes
```

Comment:
Same for 12 here.

Reply:
Yeah, it could be named something like `T_GATE_FACTORY_WITDH`